File: Tests/Tests/STPLabeledMultiFormTextFieldViewSnapshotTests.m
//
//  STPLabeledMultiFormTextFieldViewSnapshotTests.m
//  StripeiOS Tests
//
//  Created by Cameron Sabol on 3/13/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

#import <FBSnapshotTestCase/FBSnapshotTestCase.h>

#import "STPLabeledMultiFormTextFieldView.h"

#import "STPFormTextField.h"

@interface STPLabeledMultiFormTextFieldViewSnapshotTests : FBSnapshotTestCase

@end

@implementation STPLabeledMultiFormTextFieldViewSnapshotTests

//- (void)setUp {
//    [super setUp];
//
//    self.recordMode = YES;
//}

- (void)testAppearance {
    STPFormTextField *formTextField1 = [[STPFormTextField alloc] init];
    formTextField1.placeholder = @"Placeholder 1";
    formTextField1.placeholderColor = [UIColor lightGrayColor];

    STPFormTextField *formTextField2 = [[STPFormTextField alloc] init];
    formTextField2.placeholder = @"Placeholder 2";
    formTextField2.placeholderColor = [UIColor lightGrayColor];

    STPLabeledMultiFormTextFieldView *labeledFormField = [[STPLabeledMultiFormTextFieldView alloc] initWithFormLabel:@"Test Label"
                                                                                                      firstTextField:formTextField1
                                                                                                     secondTextField:formTextField2];
    labeledFormField.formBackgroundColor = [UIColor whiteColor];
    labeledFormField.frame = CGRectMake(0.f, 0.f, 320.f, 62.f);
    FBSnapshotVerifyView(labeledFormField, @"STPLabeledMultiFormTextFieldView.defaultAppearance");
}

@end

File: Tests/Tests/STPLabeledMultiFormTextFieldViewSnapshotTests.swift
//
//  STPLabeledMultiFormTextFieldViewSnapshotTests.swift
//  StripeiOS Tests
//
//  Created by Cameron Sabol on 3/13/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

import FBSnapshotTestCase

@testable import Stripe

class STPLabeledMultiFormTextFieldViewSnapshotTests: FBSnapshotTestCase {
  override func setUp() {
    super.setUp()
    //        self.recordMode = true
  }

  func testAppearance() {
    let formTextField1 = STPFormTextField()
    formTextField1.placeholder = "Placeholder 1"
    formTextField1.placeholderColor = UIColor.lightGray

    let formTextField2 = STPFormTextField()
    formTextField2.placeholder = "Placeholder 2"
    formTextField2.placeholderColor = UIColor.lightGray

    let labeledFormField = STPLabeledMultiFormTextFieldView(
      formLabel: "Test Label",
      firstTextField: formTextField1,
      secondTextField: formTextField2)
    labeledFormField.formBackgroundColor = UIColor.white
    labeledFormField.frame = CGRect(x: 0.0, y: 0.0, width: 320.0, height: 62.0)
    FBSnapshotVerifyView(
      labeledFormField, identifier: "STPLabeledMultiFormTextFieldView.defaultAppearance")
  }
}
